### ISSUE TITLE
Segmentation Fault when gbm device doesn't outlive gbm buffer object

### ISSUE THREAD

**Issue by FallingSnow** [Smithay/gbm.rs#24]
```sh
$ uname -a
Linux ayrtonslaptop 6.3.8-arch1-1 #1 SMP PREEMPT_DYNAMIC Wed, 14 Jun 2023 20:10:31 +0000 x86_64 GNU/Linux
```

https://github.com/Smithay/gbm.rs/blob/8d3cb5a6520c5f1bef667b007afee0e73f480e98/src/buffer_object.rs#L493

Is causing a segmentation fault on my system when the `gbm::Device` is dropped before the buffer object. I'm not using any unsafe code so I don't think this should be allowed to happen.

Let me know if you need any more details.

**Comment by Drakulix** [comment]
No, that shouldn't happen, see https://github.com/Smithay/gbm.rs/pull/25 for a potential solution.
